File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        pos_begin = 0 
        pos_end = len(height) -1
        max_area = 0

        while pos_end > pos_begin:
            h = min(height[pos_begin],height[pos_end])
            width = pos_end - pos_begin
            area = h * width
            max_area = max(max_area,area)

            if height[pos_begin] > height[pos_end]:
                pos_end-=1
            else:
                pos_begin+=1

        return max_area
                

        

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int posBegin = 0;
        int posEnd = height.length - 1;
        int maxArea = 0;
        int distance, h, area;
        
        while (posEnd>posBegin){
            distance = posEnd - posBegin;
            h = Integer.min(height[posBegin],height[posEnd]);
            area = h * distance;
            maxArea = Math.max(maxArea,area);

            if (height[posEnd] > height[posBegin]){
                posBegin+=1;
            }
            else{
                posEnd-=1;
            }
        }

        return maxArea;
    }
}